pkg/controller/cluster/reconciliation/cleanup_alb_reconciler.go

```diff
@@ -1,67 +0,0 @@
-package reconciliation
-
-import (
-	"fmt"
-
-	"github.com/oslokommune/okctl/pkg/cleaner"
-	"github.com/oslokommune/okctl/pkg/controller/common/reconciliation"
-
-	"github.com/oslokommune/okctl/pkg/cfn"
-
-	"github.com/oslokommune/okctl/pkg/apis/okctl.io/v1alpha1"
-
-	clientCore "github.com/oslokommune/okctl/pkg/client/core"
-
-	"github.com/oslokommune/okctl/pkg/controller/common/dependencytree"
-)
-
-type cleanupALBReconciler struct {
-	commonMetadata *reconciliation.CommonMetadata
-
-	provider v1alpha1.CloudProvider
-}
-
-// NodeType returns the relevant NodeType for this reconciler
-func (z *cleanupALBReconciler) NodeType() dependencytree.NodeType {
-	return dependencytree.NodeTypeCleanupALB
-}
-
-// SetCommonMetadata saves common metadata for use in Reconcile()
-func (z *cleanupALBReconciler) SetCommonMetadata(metadata *reconciliation.CommonMetadata) {
-	z.commonMetadata = metadata
-}
-
-// Reconcile knows how to do what is necessary to ensure the desired state is achieved
-func (z *cleanupALBReconciler) Reconcile(node *dependencytree.Node, state *clientCore.StateHandlers) (result reconciliation.Result, err error) {
-	switch node.State {
-	case dependencytree.NodeStatePresent:
-		// Nothing to do for present
-		return result, nil
-	case dependencytree.NodeStateAbsent:
-		vpc, err := state.Vpc.GetVpc(cfn.NewStackNamer().Vpc(z.commonMetadata.ClusterID.ClusterName))
-		if err != nil {
-			return result, fmt.Errorf("getting vpc: %w", err)
-		}
-
-		clean := cleaner.New(z.provider)
-
-		err = clean.DeleteDanglingALBs(vpc.VpcID)
```

Comment:
We should check if we still need this when we get delete to work as intended. Hopefully there will be no need to delete any dangling ALBs or security groups. (Out of scope for this PR though)

Reply:
Yeah. We agreed to postpone it so that the PR would be ready for review before the summer though. This reconciler is a thorn in my side, so I'd love to remove it and fix things properly at some point